cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/BasicStreamWriterSizeCollector.java

```diff
@@ -43,6 +67,16 @@ public synchronized void received(String streamName, long dataSize) {
     AtomicLong value = streamSizes.get(streamName);
     if (value == null) {
       value = streamSizes.putIfAbsent(streamName, new AtomicLong(dataSize));
+      if (value != null) {
+        // This is the first time that we've seen this stream, we subscribe to generation changes
+        truncationSubscriptions.add(streamCoordinatorClient.addListener(streamName, new StreamPropertyListener() {
```

Comment:
Would it be simpler if the collector exposes a `reset(String name)` method and let the StreamService call it (since the StreamService already subscribed to the truncation listener)?